### ISSUE TITLE
Add 'My posts' and 'Reshared' in my activity

### ISSUE THREAD

**Issue by Globulle** [diaspora/diaspora#7471]
Currently it is possible to filter 'My activity' by selecting 'Commented' or 'Liked' (see #7257). I think it should also be useful to have a filter 'My posts' (or 'Posted' or 'Published'...) within the 'My activity' menu. This way it would be easier to find something we posted some time ago.
Although this last one is less useful, I also suggest we add a 'Reshared' item. This way the menu items would be in coherence with the various possible activities in diaspora*.

**Comment by Waithamai** [comment]
What would be the difference between 'My posts' and your profile (that already lists all your posts)? 
Do you think of splitting it into own posts and reshares?

**Comment by Globulle** [comment]
> Do you think of splitting it into own posts and reshares?

Yes, I think it should be useful. And also make it accessible from "My activity". Since "My activity" exists (and remains visible on the side of the stream), it is not straighforward to access its own posts from another place (e.g. my profile). And posting definitely is an activity.

**Comment by netagonen** [comment]
Hi! I am thinking on picking up this issue, just wanted to double check this adds a new functionality. What do you think?

**Comment by Flaburgan** [comment]
We list the issues we feel important with the "Help wanted" label: https://github.com/diaspora/diaspora/labels/help%20wanted

**Comment by SuperTux88** [comment]
While this probably isn't the most requested feature, if you think it's useful and you want to work on this, you're welcome to do so (you are not forced to only do "help wanted" issues). :)

What I'm not sure here is:
* "My posts" show only status messages from me, without reshares. I think that's cool to have a way to display my own original posts.
* "Reshared" ... does this show the root posts I reshared, or my reshares? As I understand it, it should display the original posts which I reshared. But should we add a "My Reshares" too? Is any of this useful?

**Comment by SuperTux88** [comment]
And about filter between own posts and reshares: That's maybe more useful to implement on the profile (like the photos tab at the top), instead of "My activity", because it would work the for every profile, not just my own.

**Comment by goobertron** [comment]
How about adding these options as filters on the profile page stream? When viewing your profile – or that of another user – you could select 'All posts', '[original content]' or 'Reshares'.

Note: I don't think 'original content' is a good title for posts-excluding-reshares. I'll have to think of something later.

> "Reshared" ... does this show the root posts I reshared, or my reshares? As I understand it, it should display the original posts which I reshared. But should we add a "My Reshares" too? Is any of this useful?

Showing 'my' reshare of each post would be unnecessary when viewing a stream of my reshares. Displaying the root posts in each case seems more sensible.

**Comment by Globulle** [comment]
> How about adding these options as filters on the profile page stream?

I think this is not very intuitive to have *My <thing>* on one side to see what I've done, and *Profile* somewhere else which actually shows very similar things. It is certainly beyond the scope of this issue, but we may need to reorder the layout.

**Comment by SuperTux88** [comment]
I agree with @goobertron here: I think profiles should be changed to 'All posts', '[original content]' and 'Reshares'.

And then 'Reshared' could be added to My activity (and only 'Reshared' not 'My posts'), to display the original posts which I reshared. That's then consisten to 'Commented' and 'Liked' which displays posts from other people where I have done the selected action.

And profile is for what I (or the person whos profile it is) have done. I think my own posts and my own photos don't belong to the sidebar of the main stream. That's what my profile is for. And it has the advantage that I can do this for every profile, so I can display only the original posts from a person, when the person reshares a lot of things.

**Comment by netagonen** [comment]
Wow! thanks for the detailed discussion. I am still not sure I will actually pick it up but it's very helpful info for the one who will :)
